### ISSUE TITLE
[camera] [ios] Preview freezes when using torch with ultrawide or telephoto camera

### ISSUE THREAD

**Issue by Strain-solutions** [flutter/flutter#114212]
Hi.  My app requires access to the torch and a preview from the telephoto lens if using a 3-lens iPhone (e.g. iPhone 12).   When I query the camera list 

I have repeated the behaviour with a few different torch plugins so I think this is a camera issue rather than a torch issue.

## Steps to Reproduce

- if I select cameras[0] or cameras[1] (front-facing) I can generate a live preview with torch active
- If I select cameras [2] (telephoto) or [3] (ultrawide), the preview freezes while the torch is active and unfreezes when I turn the torch off. There are no error messages reported, so there is nothing obvious to 'fix' from my end.  

1. Execute `flutter run` on the code sample 
2. Select a camera by changing the index value in onNewCameraSelected in initState method.
3. Toggle the torch using the text button

**Expected results:** Preview should not freeze for any cameras

**Actual results:** Preview freezes for cameras 2 and 3 but does not freeze for camera 0 and 1

## code snippet

````
import 'package:camera/camera.dart';
import 'package:flutter/material.dart';
import 'package:torch_controller/torch_controller.dart';

List<CameraDescription> cameras = [];

void main() async {
  TorchController().initialize();
  WidgetsFlutterBinding.ensureInitialized();
  cameras = await availableCameras();

  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Flutter Demo',
      theme: ThemeData(
        // enable Material 3
        useMaterial3: true,
        colorScheme: const ColorScheme.light(
          primary: Colors.indigo,
          outline: Colors.indigo,
        ),
      ),
      home: const CameraTest(),
    );
  }
}

class CameraTest extends StatefulWidget {
  const CameraTest({Key? key}) : super(key: key);

  @override
  _CameraTestState createState() => _CameraTestState();
}

class _CameraTestState extends State<CameraTest> with WidgetsBindingObserver {
  double? previewHt;
  CameraController? controller;
  bool _isCameraInitialized = false;
  TorchController? tc;



  void onNewCameraSelected(CameraDescription cameraDescription) async {
    final previousCameraController = controller;
    // Instantiating the camera controller
    final CameraController cameraController = CameraController(
      cameraDescription,
      ResolutionPreset.high,
      imageFormatGroup: ImageFormatGroup.bgra8888,
    );

    // Dispose the previous controller
    await previousCameraController?.dispose();

    // Replace with the new controller
    if (mounted) {
      setState(() {
        controller = cameraController;
      });
    }

    // Update UI if controller updated
    cameraController.addListener(() {
      if (mounted) setState(() {});
    });

    // Initialize controller
    try {
      await cameraController.initialize();
    } on CameraException catch (e) {
      print('Error initializing camera: $e');
    }
    // Update the Boolean
    if (mounted) {
      setState(() {
        _isCameraInitialized = controller!.value.isInitialized;
      });
    }
  }

  @override
  void initState() {
    onNewCameraSelected(cameras[2]); // < change to 0 or 1 and preview is fine
    tc = TorchController();
    super.initState();
  }

  @override
  void dispose() {
    controller?.dispose();
    super.dispose();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state) {
    final CameraController? cameraController = controller;

    // App state changed before we got the chance to initialize.
    if (cameraController == null || !cameraController.value.isInitialized) {
      return;
    }

    if (state == AppLifecycleState.inactive) {
      // Free up memory when camera not active
      cameraController.dispose();
    } else if (state == AppLifecycleState.resumed) {
      // Reinitialize the camera with same properties
      onNewCameraSelected(cameraController.description);
    }
  }

  @override
  Widget build(BuildContext context) {
    previewHt = MediaQuery.of(context).size.height*.7;

    return Scaffold(
        body: Column(
          children: [
            SizedBox(
              height: previewHt,
              child: _isCameraInitialized
                  ? AspectRatio(
                aspectRatio: 1 / controller!.value.aspectRatio,
                child: controller!.buildPreview(),
              )
                  : Container(),
            ),
            Center(
              child: TextButton(
                child: const Text('Toggle Torch',
                  style: TextStyle(fontSize: 20),),
              onPressed: ()=>tc!.toggle(),),
            )
          ],
        )
    );
  }
}

````

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1507, on macOS 12.2.1 21D62 darwin-arm, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.71.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!

```

</details>

**Comment by danagbemava-nc** [comment]
Hi @Strain-solutions, I do not have an iOS device with an ultra-wide camera as such I cannot verify this locally. Can you share a recording of the phenomena so it's easier for us to visualise the issue? 

Thank you

**Comment by Strain-solutions** [comment]
HI sorry for the delay.  I was away from my desk over the weekend.  Ive attached a screen recording using the above code and choosing camera [3].  You can see from the recording that the preview initially works fine.  At around 5 sec in the 'toggle torch" button is pressed an the preview freezes as the torch comes on.  A few seconds later "toggle torch" is pressed again, the torch goes off and the preview resumes. 

If you need more information or a longer video please let me know.

https://user-images.githubusercontent.com/58543579/198937983-fdc5bbcc-3a80-457e-ac80-fb1c6c7d2d23.mp4

**Comment by Strain-solutions** [comment]
HI, sorry for the delay.  I was away from my desk over the weekend.  I've attached a screen recording using the above code and choosing camera [3].  

You can see from the recording that the preview initially works fine.  

At around 5 sec the 'toggle torch" button is pressed, and the preview freezes as the torch comes on (to be clear, I am still moving the phone from side to side at this point).  

A few seconds later "toggle torch" is pressed again, the torch goes off, and the preview resumes. 

If you need more information or a longer video, please let me know.


https://user-images.githubusercontent.com/58543579/198938916-311e4c15-d504-49bc-a557-3c1e2f83800c.mp4

**Comment by danagbemava-nc** [comment]
Thanks for the video. 

Labeling this for further investigation as I do not have the device required to reproduce this issue locally.

**Comment by hellohuanlin** [comment]
This could be a limitation on iOS - when I launch the Apple's camera app, the torch option becomes unavailable. So you may want to do the same and disable the torch when previewing the video.

**Comment by Strain-solutions** [comment]
Thanks, @hellohuanlin. I don't think that's it (also, I specifically need both torch and preview together).  Anyway, I think this is some limitation with how the camera package interacts with other packages (or perhaps more likely how they interact with the camera plugin); I'll try to explain for other poor souls who may find this thread, but the long and short of it is you can probably close this issue.

This is an old project I've come back to, and at the time I initially wrote the code, the camera controller in the camera package did not have a `controller.setFlashMode(Torch)` option (which it does now), and so I used a different torch package.  If you look at the code above, I use the torch controller package.  That package and a couple of others I have tried give the behaviour outlined in this thread (the torch turns off when the preview is activated).  However, if I use the `controller.setFlashMode(Torch)`, everything works as required  

TLDR: activate the torch using the camera package controller and not another package. 

I've not closed it from my end in case you need to look any further into how the different packages interact, but Im happy for this to be closed.  

This is actually my first bug report to the flutter team so thanks for taking it seriously!
